wasm/instructions.h

```diff
@@ -23,29 +26,72 @@ struct BlockType {
     };
     std::variant<Empty, ValueType, TypeIdx> value;
     [[nodiscard]] bool operator==(BlockType const &) const = default;
+
+    constexpr std::string to_string() {
```

Comment:
Any particular reason you made this a member instead of a free function? It might be nice to maintain the same `constexpr std::string to_string(T const &)` signature that the stdlib uses for all to_string-things?

Reply:
I thought maybe it seemed cleaner, but I think I like this style just as well.